enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/UnsecuredAPIAuthenticator.java

```diff
@@ -79,7 +86,29 @@ public AuthenticationContext authenticate(RequestContext requestContext) throws
                 throw new APISecurityException(APIConstants.StatusCodes.SERVICE_UNAVAILABLE.getCode(),
                         GeneralErrorCodeConstants.API_BLOCKED_CODE, GeneralErrorCodeConstants.API_BLOCKED_MESSAGE);
             }
-            return FilterUtils.generateAuthenticationContextForUnsecured(requestContext);
+
+            // Generate or get backend JWT
+            JWTConfigurationDto backendJwtConfig = ConfigHolder.getInstance().getConfig().
+                    getJwtConfigurationDto();
+            String endUserToken = null;
+
+            if (backendJwtConfig.isEnabled()) {
+                JWTValidationInfo validationInfo = new JWTValidationInfo();
+
+                APIKeyValidationInfoDTO apiKeyValidationInfoDTO = new APIKeyValidationInfoDTO();
+                apiKeyValidationInfoDTO.setApiName(requestContext.getMatchedAPI().getName());
+                apiKeyValidationInfoDTO.setApiVersion(requestContext.getMatchedAPI().getVersion());
+
+                AbstractAPIMgtGatewayJWTGenerator jwtGenerator = BackendJwtUtils.getApiMgtGatewayJWTGenerator();
+                boolean isGatewayTokenCacheEnabled = ConfigHolder.getInstance().getConfig().getCacheDto().isEnabled();
+                JWTInfoDto jwtInfoDto = FilterUtils.generateJWTInfoDto(null, validationInfo,
+                        apiKeyValidationInfoDTO, requestContext);
+                endUserToken = BackendJwtUtils.generateAndRetrieveJWTToken(jwtGenerator, "",
```

Comment:
As per [this](https://github.com/wso2/product-microgateway/blob/5561ee9685c7e1c794a97eeed6e5a9178fbca77c/enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/util/BackendJwtUtils.java#L78) having the token signature as "" would mean that the same Backend JWT would be sent for requests by all users. Which means jti, iat, exp all will be the same for all users. Suggesting to have a different approach. Let's reconsider about generating JWTs for the insecure case, because nothing gets validated.

Reply:
@praminda